source/schema/avro.go

```diff
@@ -0,0 +1,161 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"cmp"
+	"fmt"
+	"math"
+	"slices"
+
+	"github.com/hamba/avro/v2"
+	"github.com/jackc/pglogrepl"
+	"github.com/jackc/pgx/v5/pgconn"
+	"github.com/jackc/pgx/v5/pgtype"
+)
+
+const (
+	avroNS = "conduit.postgres"
+)
+
+var Avro = &avroExtractor{
+	pgMap: pgtype.NewMap(),
+	avroMap: map[string]*avro.PrimitiveSchema{
```

Comment:
It looks like, at least for now, we don't care if a column is not nullable?

Reply:
Yes, all these are right now "required" I guess. Nullability will depend on the pg data type, which I have not added. Is this what you mean?

Reply:
Yup that's what I meant.